.github/workflows/deploy.yaml

```diff
@@ -24,6 +24,20 @@ env:
   IMAGE_NAME: ${{ github.repository }}
 
 jobs:
+  lint:
+    name: Lint Markdown with Vale
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Run Vale linting
+        uses: errata-ai/vale-action@v2.1.1
```

Comment:
I would like to specify the image by hash instead of the version to prevent supply chain attacks from a compromised dependency of external Github Action image.

Reference:
https://github.com/GoogleCloudPlatform/khi/blob/main/.github/workflows/pr-request.yaml#L94-L97